### PR TITLE
PFSA-903

### PR DESCRIPTION
**Pull request by ezengsvmx**
@ericsvmx @NeerajNemaSVMX Can you help review?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,26 @@
...
+
+Steps:
+1. curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.2/install.sh | bash
+2. Open a new shell
+3. nvm install v6.9.4 # Install Node 6.9.4
+4. nvm use 6.9.4
+5. npm install -g react-native-cli
+6. npm install
+7. sudo launchctl remove com.mcafee.agent.macmn # Stop the virus scanner on MacBook
+8. Running on Android
```
Comment (NeerajNemaSVMX) [inline README.md]: Please add one step to kill the McAffee agent before Running on any device.

Comment (NeerajNemaSVMX) [inline README.md]: @ezengsvmx

File: js/common.js [hunk 2]
```diff
@@ -15,14 +15,30 @@ export default class SVMXExternalApp extends Component {
     super(props);
     this.state = {text: ''};
     this.state = {selectedTab: 'home' };
+    this.state = {receivedText: ''};
```
Comment (ericsvmx) [inline js/common.js]: This is resetting the `state` object on each call. It should be like this:
```
this.state = {
  text: '',
  selectedTab: 'home,
  receivedText: '',
};
```

File: js/common.js [hunk 3]
```diff
@@ -15,14 +15,30 @@ export default class SVMXExternalApp extends Component {
...
   }
 
   componentDidMount() {
+    this.setState({selectedTab: 'home' });
```
Comment (ericsvmx) [inline js/common.js]: By convention, add spaces inside an object, like this:
```
this.setState({ selectedTab: 'home' });
```

File: js/common.js [hunk 4]
```diff
@@ -15,14 +15,30 @@ export default class SVMXExternalApp extends Component {
...
 
-  _handleOpenURL(event) {
+  _handleOpenURL = (event) => {
     console.log(event.url);
+    try {
+      var equalSign = event.url.indexOf('=') + 1;
+      var jsonString = decodeURI(event.url.substring(equalSign, event.url.length));
+      var json = JSON.parse(jsonString);
+      var indent = 2;
+      var prettyJson = JSON.stringify(json, undefined, indent);
```
Comment (ericsvmx) [inline js/common.js]: Always use `const` instead of `var`, unless a variable needs to be re-assigned and then use `let`.

File: js/common.js [hunk 5]
```diff
@@ -94,17 +110,24 @@ export default class SVMXExternalApp extends Component {
 
         <TabNavigator.Item
           titleStyle={styles.welcome}
-          selected={this.state.selectedTab === 'profile'}
+          selected={this.state.selectedTab === 'received'}
           title="Receive JSON Data"
-          onPress={() => this.setState({ selectedTab: 'profile' })} >
+          onPress={() => this.setState({ selectedTab: 'received' })} >
```
Comment (ericsvmx) [inline js/common.js]: As a best practice always define event handlers in the class and not during render. Performance reasons. For example:
```
...
onPressTabNavigatorItem = () => {
  this.setState({ selectedTab: event.currentTarget.dataset.key });
};
...
```
And use `data-key` type attributes on elements to pass data as such.

File: js/common.js [hunk 6]
```diff
@@ -94,17 +110,24 @@ export default class SVMXExternalApp extends Component {
...
-              <Text style = {styles.normaltext}>
                 Received data
-              </Text>
             </Text>
+            <TextInput
+              style={{height: "60%", borderColor: 'gray', borderWidth: 10, fontSize: 20, margin: 20, borderRadius: 10, borderWidth: 2}}
+              multiline = {true}
+              editable = {false}
+              numberOfLines = {4}
+              borderColor = 'gray'
```
Comment (ericsvmx) [inline js/common.js]: Inconsistent white space. No spaces between ` = ` here.

**Comment by ezengsvmx** [comment]
@ericsvmx Thanks for the inputs, I will work on them.

BTW, is there a linter in your team to catch the code formatting issues?

File: README.md (hunk 1 above)
Comment (ezengsvmx) [inline README.md]: See step 7

File: js/common.js (hunk 2 above)
Comment (ezengsvmx) [inline js/common.js]: Done.

File: js/common.js (hunk 3 above)
Comment (ezengsvmx) [inline js/common.js]: Done

File: js/common.js (hunk 4 above)
Comment (ezengsvmx) [inline js/common.js]: Done

File: js/common.js (hunk 5 above)
Comment (ezengsvmx) [inline js/common.js]: Done

File: js/common.js (hunk 6 above)
Comment (ezengsvmx) [inline js/common.js]: Done

**Pull request closed without merging**